examples/pennylane_run_rotoselect.py

```diff
@@ -0,0 +1,432 @@
+    params[d] = -np.pi / 2.0 - a
+    # restrict output to lie in (-pi,pi], a convention
+    # consistent with the Rotosolve paper
+    if params[d] <= -np.pi:
+        params[d] += 2*np.pi
+
+
+# one cycle of rotosolve
+def rotosolve_cycle(cost, params):
+    for d in range(len(params)):
+        opt_theta(d, params, cost)
+    return params
+
+
+##############################################################################
+# Optimization and comparison with gradient descent
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# We set up some initial parameters for the :math:`R_x` and :math:`R_y`
+# gates in the ansatz circuit structure and perform an optimization using the
+# Rotosolve algorithm.
+
+init_params = [0.3, 0.25]
+params = init_params.copy()
+n_steps = 30
+
+costs_rotosolve = []
+
+for i in range(n_steps):
+    costs_rotosolve.append(cost(params))
+    params = rotosolve_cycle(cost, params)
+
+##############################################################################
+# We then compare the results of Rotosolve to an optimization
+# performed with gradient descent and plot
+# the cost functions at each step (or cycle in the case of Rotosolve).
+# This comparison is fair since the number of circuit
+# evaluations involved in a cycle of Rotosolve is similar to those required to calculate
+# the gradient of the circuit and step in this direction. Evidently, the Rotosolve algorithm
+# converges on the minimum after the first cycle for this simple circuit.
+
+params = init_params.copy()
+opt = qml.GradientDescentOptimizer(stepsize=0.5)
+costs_grad_desc = []
+for i in range(n_steps):
+    costs_grad_desc.append(cost(params))
+    params = opt.step(cost, params)
+
+
+# plot cost function optimization using the 2 techniques
+import matplotlib.pyplot as plt
+
+steps = np.arange(0, n_steps)
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(7, 3))
+plt.subplot(1, 2, 1)
+plt.plot(steps, costs_grad_desc, "o-")
+plt.title("grad. desc.")
+plt.xlabel("steps")
+plt.ylabel("cost")
+plt.subplot(1, 2, 2)
+plt.plot(steps, costs_rotosolve, "o-")
+plt.title("rotosolve")
+plt.xlabel("cycles")
+plt.ylabel("cost")
+plt.tight_layout()
+plt.show()
+
+
+##############################################################################
+# Cost function surface for circuit ansatz
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Now, we plot the cost function surface for later comparison with the surface generated
+# by learning the circuit structure. It is apparent that, based on the structure
+# ansatz chosen above, the cost function does not depend on the angle parameter :math:`\theta_2`
+# for the rotation gate :math:`R_y`. As we will show in the following sections, this independence is not true
+# for alternative gate choices.
+
+from matplotlib import cm
+from matplotlib.ticker import MaxNLocator
+from mpl_toolkits.mplot3d import Axes3D
+
+fig = plt.figure(figsize=(6, 4))
+ax = fig.gca(projection="3d")
+
+X = np.linspace(-4.0, 4.0, 20)
```

Comment:
let's throw a few more points in the `theta_1` axis. Figure could look a big smoother

Reply:
Done

Reply:
I don't see any changes in git yet

Reply:
Oh I changed the number of points for the last surface and not this one, changing it now.